### PR TITLE
add whitelist per_address_limit

### LINKED ISSUE #79: `per_address_limit` for whitelist

### PR DESCRIPTION
**Pull request by yubrew**
fix #79

whitelist has `per_address_limit` for minting. this allows creators to have a portion to whitelist community, and a portion to public mint

### CODE REVIEW

File: contracts/whitelist/src/contract.rs [hunk 1]
```diff
@@ -47,6 +48,7 @@ pub fn instantiate(
         end_time: msg.end_time,
         num_members: msg.members.len() as u32,
         unit_price: msg.unit_price,
+        per_address_limit: msg.per_address_limit,
```
Comment (jhernandezb) [inline contracts/whitelist/src/contract.rs:51]: lets check limit is > 0

**Review by jhernandezb — COMMENTED** [review]
can the admin update the limit after instantiated?

File: contracts/whitelist/src/contract.rs [hunk 2]
```diff
@@ -22,6 +23,7 @@ const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 const MAX_MEMBERS: u32 = 5000;
 const CREATION_FEE: u128 = 100_000_000;
 const MIN_MINT_PRICE: u128 = 25_000_000;
+const MAX_PER_ADDRESS_LIMIT: u64 = 30;
```
Comment (shanev) [inline contracts/whitelist/src/contract.rs:26]: If max members is a `u32`, then this type doesn't have to be larger than it right?

**Review by shanev — CHANGES_REQUESTED** [review]

File: contracts/whitelist/src/contract.rs (hunk 2 above)
Comment (yubrew) [reply contracts/whitelist/src/contract.rs]: you're right, changed to `u32`

File: contracts/whitelist/src/contract.rs (hunk 1 above)
Comment (yubrew) [reply contracts/whitelist/src/contract.rs:66]: 👍

**Comment by yubrew** [comment]
> can the admin update the limit after instantiated?

yes added this

File: contracts/whitelist/src/contract.rs [hunk 3]
```diff
@@ -41,12 +43,27 @@ pub fn instantiate(
...
     }
 
+    // Check per address limit is valid
+    if msg.per_address_limit > MAX_PER_ADDRESS_LIMIT {
+        return Err(ContractError::InvalidPerAddressLimit {
+            max: MAX_PER_ADDRESS_LIMIT.to_string(),
+            got: msg.per_address_limit.to_string(),
+        });
+    }
+    if msg.per_address_limit == 0 {
```
Comment (jhernandezb) [inline contracts/whitelist/src/contract.rs:53]:
```suggestion
    if msg.per_address_limit > 0 {
```

**Review by shanev — APPROVED** [review]

**Pull request merged**